api/src/main/scala/org/alephium/api/model/TimeInterval.scala

```diff
@@ -16,6 +16,19 @@
 
 package org.alephium.api.model
 
-import org.alephium.util.TimeStamp
+import sttp.model.StatusCode
 
-final case class TimeInterval(from: TimeStamp, to: TimeStamp)
+import org.alephium.api.ApiError
+import org.alephium.util.{Duration, TimeStamp}
+
+final case class TimeInterval(from: TimeStamp, to: TimeStamp) {
+  def validateTimeSpan(max: Duration): Either[ApiError[_ <: StatusCode], Unit] = {
+    if (durationUnsafe() > max) {
+      Left(ApiError.BadRequest(s"Time span cannot be greater than ${max}"))
+    } else {
+      Right(())
+    }
+  }
+
+  def durationUnsafe(): Duration = to.deltaUnsafe(from)
```

Comment:
we need to make sure `durationUnsafe() > 0`, maybe we can modify the condition to `timeInterval.fromTs >= timeInterval.toTs` at [timeIntervalQuery](https://github.com/alephium/alephium/blob/e69b108e6028ea860afc7e73bcae3bc2efd21a08/api/src/main/scala/org/alephium/api/Endpoints.scala#L56)?

Reply:
Nice catch! I will fix it